db_handler.py

```diff
@@ -0,0 +1,34 @@
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker, query, Query
+from contextlib import contextmanager
+
+Base = declarative_base()
+engine = create_engine('sqlite:///midas.db', echo=True)
+Session = sessionmaker(bind=engine)
+
+
+def create_all_tables():
+    Base.metadata.create_all(engine)
+
+
+def recreate_all_tables():
+    Base.metadata.drop_all(engine)
+    Base.metadata.create_all(engine)
+
+
+class Refiner(Query):
+    def __init__(self, entities, session=None):
+        super().__init__(entities, session)
+
+    def refine(self, *args, **kwargs):
+        if len(kwargs) == 0:
+            return self.filter(*args)
```

Comment:
You also need to support chaining of calls to `refine`, (Midas4.3):
```midas.Organization.get().refine(name='PASTEN').refine(Organization.prime_location == 'ASDF')```

Reply:
It works as implemented. Added a test to verify